### ISSUE TITLE
New rule proposal: "Operands Must Be On Same Line Or Separate Lines"

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#752]
This is similar to SA1117 (ParametersMustBeOnSameLineOrSeparateLines), but it applies to the operands of a chain of binary expressions. It is a complement but otherwise unrelated to #748.

**Category:** Readability
**Rule:** Operands must be on same line or separate lines
**Description:** For a sequence of binary expressions, e.g. `x && y && z`, each operand must be on the same line, or each operand must be on a line by itself.

**Example:**

```cs
// allowed:
x && y && z

// allowed:
x
&& y
&& z

// not allowed:
x && y
&& z
```

**Comment by Przemyslaw-W** [comment]
I like this. However, rule should take "nesting level" into account:
```
// allowed
(x && y) 
|| (z && v)
```

**Comment by sharwell** [comment]
That's more of a "note" than a "however". There are three different binary expressions here, each of which does meet the rule.

**Comment by GregReddick** [comment]
:+1:
